rdr_service/dao/bq_questionnaire_dao.py

```diff
@@ -47,45 +104,91 @@ def make_bqrecord(self, p_id, module_id, latest=False, convert_to_enum=False):
             logging.info('Generator: ignoring questionnaire module id [{0}].'.format(module_id))
             return None, list()
 
-        qnans = self.ro_dao.call_proc('sp_get_questionnaire_answers', args=[module_id, p_id])
-        if not qnans or len(qnans) == 0:
-            return None, list()
-
-        bqrs = list()
-        for qnan in qnans:
-            bqr = BQRecord(schema=table().get_schema(), data=qnan, convert_to_enum=convert_to_enum)
-            bqr.participant_id = p_id  # reset participant_id.
-
-            fields = bqr.get_fields()
-            for field in fields:
-                fld_name = field['name']
-                if fld_name in (
-                    'id',
-                    'created',
-                    'modified',
-                    'authored',
-                    'language',
-                    'participant_id',
-                    'questionnaire_response_id'
-                ):
-                    continue
-
-                fld_value = getattr(bqr, fld_name, None)
-                if fld_value is None:  # Let empty strings pass.
-                    continue
-                # question responses values need to be coerced to a String type.
-                if isinstance(fld_value, (datetime.date, datetime.datetime)):
-                    setattr(bqr, fld_name, fld_value.isoformat())
-                else:
-                    setattr(bqr, fld_name, str(fld_value))
-
-                # Truncate zip codes to 3 digits
-                if fld_name in ('StreetAddress_PIIZIP', 'EmploymentWorkAddress_ZipCode') and len(fld_value) > 2:
-                    setattr(bqr, fld_name, fld_value[:3])
-
-            bqrs.append(bqr)
-            if latest:
-                break
+        # This section refactors how the module answer data is built, replacing  the sp_get_questionnaire_answers
```

Comment:
Note on testing:
During development, I would use the resource rebuild-pids tool to exercise the new code, and build the questionnaire data both using the new SQL logic and the old stored procedure call in the same make_bqrecord() invocation.   The resulting bqr records (new vs. old) were compared field by field to make sure they had matching answer values. 

This would verify the new logic is correctly replicating exactly what the stored procedure would for other modules (Basics, LifeStyle, OverallHealth, etc.), and how I caught mismatches such as the new logic not initially handling multi-select answers.

That gave me confidence the refactored code should be producing results consistent with our existing pdr_mod_* data.